File: app/src/main/java/com/example/pam_project/database/lists/ListEntity.java
package com.example.pam_project.database.lists;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.example.pam_project.database.categories.CategoryEntity;

import static androidx.room.ForeignKey.CASCADE;

@Entity(tableName = "Lists", foreignKeys = @ForeignKey(entity = CategoryEntity.class,
        parentColumns = "id",
        childColumns = "categoryId",
        onDelete = CASCADE))
public class ListEntity {

    @PrimaryKey(autoGenerate = true)
    public long id;

    @ColumnInfo(name = "name")
    public final String name;

    @ColumnInfo(name = "categoryId")
    public final long categoryId;

    public ListEntity(long id, String name, long categoryId) {
        this.id = id;
        this.name = name;
        this.categoryId = categoryId;
    }

    @Ignore
    public ListEntity(String name, long categoryId) {
        this.name = name;
        this.categoryId = categoryId;
    }
}

File: app/src/main/java/com/example/pam_project/database/lists/ListEntity.kt
package com.example.pam_project.database.lists

import androidx.room.*
import com.example.pam_project.database.categories.CategoryEntity

@Entity(tableName = "Lists")
class ListEntity {
    @kotlin.jvm.JvmField
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @kotlin.jvm.JvmField
    @ColumnInfo(name = "name")
    val name: String

    @kotlin.jvm.JvmField
    @ForeignKey(entity = CategoryEntity::class, parentColumns = ["id"], childColumns = ["categoryId"], onDelete = ForeignKey.CASCADE)
    val categoryId: Long

    constructor(id: Long, name: String, categoryId: Long) {
        this.id = id
        this.name = name
        this.categoryId = categoryId
    }

    @Ignore
    constructor(name: String, categoryId: Long) {
        this.name = name
        this.categoryId = categoryId
    }
}